contrib/mesos/pkg/executor/service/service.go

```diff
@@ -78,15 +87,31 @@ func (s *KubeletExecutorServer) runExecutor(
 	apiclient *clientset.Clientset,
 	registry executor.Registry,
 ) (<-chan struct{}, error) {
+	staticPodFilters := podutil.Filters{
+		// annotate the pod with BindingHostKey so that the scheduler will ignore the pod
+		// once it appears in the pod registry. the stock kubelet sets the pod host in order
+		// to accomplish the same; we do this because the k8sm scheduler works differently.
+		podutil.Annotator(map[string]string{
+			meta.BindingHostKey: s.HostnameOverride,
```

Comment:
pretty sure that the `HostnameOverride` here is the same as the agent name that was being used from the `SlaveInfo` during registration. Need to double-checkl